src/plugins/elasticsearch/lib/expose_client.js

```diff
@@ -78,8 +78,8 @@ module.exports = function (server) {
   server.expose('ElasticsearchClientLogging', ElasticsearchClientLogging);
   server.expose('client', client);
   server.expose('createClient', createClient);
-  server.expose('callWithRequestFactory', callWithRequest);
-  server.expose('callWithRequest', callWithRequest(noAuthClient));
+  server.expose('callWithRequestFactory', _.partial(callWithRequest, server));
```

Comment:
@lukasolson Hey, I know you originally introduced `callWithRequestFactory` here in https://github.com/elastic/kibana/commit/5be340079b81cf934744159fc3aaa2af56d55747 so I was wondering if you could take a look at my refactor here. Let me know if you think there's a better way to do this.

BTW, I noticed that no code is actually using `callWithRequestFactory`. Is that expected?